translate/tracer.go

```diff
@@ -28,14 +42,66 @@ type MemoryTable struct {
 	MemoryCounter uint
 }
 
-// TODO: implement me w/ test cases :-)
+type MemoryTableDefiner interface {
+	Define(id string, value interface{}) (MemoryTableEntry, error)
```

Comment:
we separate `Definer` and `Getter`?

caller of `MemoryTable` separated?

Reply:
Because of the go style interface which has only one method. I’ll combine them!